src/browserlib/extract-events.mjs

```diff
@@ -0,0 +1,278 @@
+import informativeSelector from './informative-selector.mjs';
+import extractWebIdl from './extract-webidl.mjs';
+import {parse} from "../../node_modules/webidl2/index.js";
+import getAbsoluteUrl from './get-absolute-url.mjs';
+
+const isSameEvent = (e1, e2) => (e1.href && e1.href === e2.href) || (e1.type === e2.type && e1.targets?.sort()?.join("|") === e2.targets?.sort()?.join("|"));
+
+const singlePage = !document.querySelector('[data-reffy-page]');
+
+const href = el => getAbsoluteUrl(el || document.body.querySelector("*[id]"), {singlePage});
+
+export default function (spec) {
+  // Used to find eventhandler attributes
+  const idl = extractWebIdl();
+  const idlTree = parse(idl);
+  const idlInterfaces = idlTree.filter(item => item.type === "interface" || item.type === "interface mixin");
+
+  // associate event names from event handlers to interfaces with such an handler
+  const handledEventNames = idlInterfaces.map(iface => iface.members.filter(m => m.idlType?.idlType === "EventHandler" && m.type === "attribute" && m.name?.startsWith("on")).map(m => [m.name.slice(2), iface.name])).flat().reduce((acc, b) => {
+    if (!acc[b[0]]) acc[b[0]] = [];
+    acc[b[0]].push(b[1]);
+    return acc;
+  }, {});
+
+
+  function fromEventElementToTargetInterfaces(eventEl) {
+    if (!eventEl) return;
+
+    if (eventEl.dataset?.dfnFor || eventEl.dataset?.linkFor) {
+      return (eventEl.dataset.dfnFor || eventEl.dataset.linkFor).split(",").map(t => t.trim());
+    } else if (eventEl.getAttribute("href")?.startsWith("#")) {
+      const dfn = document.getElementById(eventEl.getAttribute("href").slice(1));
+      if (dfn && dfn.dataset?.dfnFor) {
+	return dfn.dataset.dfnFor.split(",").map(t => t.trim());
+      }
+    } else if (handledEventNames[eventEl.textContent]?.length) {
+      // Search for on<event> EventHandler in IDL
+    const matchingInterfaces = handledEventNames[eventEl.textContent];
+      if (matchingInterfaces.length === 1) {
+	// only one such handler, we assume it's a match
+	return matchingInterfaces;
+      } else {
+	console.error("[reffy] Multiple event handler named " + eventEl.textContent + ", cannot associate reliably to an interface in " + spec.title);
```

Comment:
Errors will be logged but we'll likely happily ignore them. Should this (and other occurrences that report an error) throw and require spec to be patched instead?

Reply:
I think I would prefer job failures through webref package guarantees rather than through reffy errors; in particular, there may be a number of cases where patching the extracts is a more likely (and certainly a more timely) outcome than getting the spec patched (esp. since in a number of cases, it won't be something that can easily be qualified as a fix).